src/pocketmine/item/Food.php

```diff
@@ -41,7 +42,14 @@ public function getAdditionalEffects() : array{
 		return [];
 	}
 
-	public function onConsume(Living $consumer){
+	public function canBeConsumedBy(Living $consumer): bool{
+		return ($this->requiresHunger() and $consumer instanceof Human and !$consumer->isHungry()) ? false : true;
+	}
 
+	public function onConsume(Living $consumer){
+		if($consumer instanceof Human) {
+			$consumer->addFood($this->getFoodRestore());
+			$consumer->addSaturation($this->getSaturationRestore());
+		}
```

Comment:
This should be an event, in my opinion. However I do see the `canBeConsumedBy()` pretty useless as one, mobs aren't implemented with functionality, two, if they were they wouldn't be triggering this event. (At least to my knowledge). 
I also know that PM is heavily based on the Bukkit api, and this event is not present for entities there either (http://bukkit.luricos.de/api/1.7.9-R0.2/d9/da5/namespaceorg_1_1bukkit_1_1event_1_1entity.html).